src/Dropdown/Dropdown.spec.js

```diff
@@ -24,3 +42,79 @@ test('Dropdown', () => {
 
   expect(component.toJSON()).toMatchSnapshot();
 });
+
+triggerEvents.forEach( event => {
+
+test(`it opens on toggle event: ${event}`, () => {
+  const stopPropagation = jest.fn();
+  const wrapper = shallow(<Dropdown theme={defaultTheme} trigger={<Button>Trigger</Button>} on={event}>
+      {() => (
+        <Fragment>
+          <Dropdown.Header>Test</Dropdown.Header>
+          <Dropdown.Body>Body</Dropdown.Body>
+          <Dropdown.Footer>Footer</Dropdown.Footer>
+        </Fragment>
+      )}
+    </Dropdown>);
+
+  // open on trigger
+  const trigger = wrapper.childAt(0);
+  trigger.simulate(event, { type: event, stopPropagation });
+  expect(wrapper).toMatchSnapshot()
+})
+
+const blurEvent = blurEvents[event];
+const focusEvent = focusEvents[event];
+
+test(`closes on ${blurEvents[event]} event`, async done => {
+  const stopPropagation = jest.fn();
+  const wrapper = shallow(<Dropdown theme={defaultTheme} trigger={<Button>Trigger</Button>} on={event}>
+      {() => (
+        <Fragment>
+          <Dropdown.Header>Test</Dropdown.Header>
+          <Dropdown.Body>Body</Dropdown.Body>
+          <Dropdown.Footer>Footer</Dropdown.Footer>
+        </Fragment>
+      )}
+    </Dropdown>);
+  
+  // open on trigger
+  const trigger = wrapper.childAt(0);
+  trigger.simulate(event, { type: event, stopPropagation });
+
+  // blur dropdown
+  wrapper.simulate(blurEvent, { type: blurEvent, stopPropagation });
+  await setTimeout(() => {
```

Comment:
nit: not sure `await setTimeout` does what you want and you're using the `done` callback anyways

Reply:
Good point, removed.